compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -599,12 +600,12 @@ pub fn create_compressed_metadata_file_for_xcoff(
         section: SymbolSection::Section(data_section),
         flags: SymbolFlags::None,
     });
-    let len = data.len() as u64;
-    let offset = file.append_section_data(section, &len.to_le_bytes(), 1);
+    let len = data.len() as u32;
```

Comment:
The length of `data` here is not guaranteed to be less than `u32::MAX`. Ideally we'd produce a compile error here, but it's not clear to me if we have access to a `TyCtxt` to do that. At the very least we should use `.try_into().unwrap()` to ICE early instead of producing a truncated metadata section.

Reply:
At the very least there is a session: https://github.com/rust-lang/rust/blob/14e6f3f5625fb719cb337b21722c88cbbc85f884/compiler/rustc_codegen_ssa/src/back/metadata.rs#L535

Reply:
@bzEq even though we panic above, this should also be `.try_into().unwrap()`